jellyfin-api/src/commonMain/kotlin/org/jellyfin/sdk/api/sockets/SocketInstance.kt

```diff
@@ -174,36 +176,39 @@ public class SocketInstance internal constructor(
 			return
 		}
 
-		connection = socketConnectionFactory.create(
-			api.httpClientOptions,
-			incomingMessages,
-			outgoingMessages,
-			coroutineContext,
-		).apply {
-			val connected = connect(UrlBuilder.buildUrl(
-				baseUrl = requireNotNull(baseUrl),
-				pathTemplate = SOCKET_URL,
-				queryParameters = mapOf(
-					QUERY_DEVICE_ID to deviceInfo.id,
-					QUERY_ACCESS_TOKEN to accessToken,
-				)
-			).replace(Regex("^http"), "ws"))
-
-			if (connected) {
-				messageForwardJob?.cancel()
-				messageForwardJob = coroutineScope.launch {
-					for (message in incomingMessages) forwardMessage(message)
-				}
-				listenerHelper.activeSubscriptions.clear()
-				updateSubscriptions()
-				_state.value = SocketInstanceState.CONNECTED
-			} else {
-				_state.value = SocketInstanceState.ERROR
-				// TODO retry: failed connecting
-			}
+		connectionScope = coroutineScope + Job()
+		connectionScope!!.launch {
+			connection = socketConnectionFactory.create(
+				api.httpClientOptions,
+				incomingMessages,
+				connectionScope!!.coroutineContext,
+			).connectAndBind(connectionScope!!)
```

Comment:
Another location where an extra variable would be preferable to all the `!!`.

Reply:
Something like this?

```kotlin
connectionScope = (coroutineScope + Job()).also { connectionScope ->
	connectionScope.launch {
		connection = socketConnectionFactory.create(
			api.httpClientOptions,
			incomingMessages,
			connectionScope.coroutineContext,
		).connectAndBind(connectionScope)
	}
}
```

Reply:
Or like this

```kotlin
(coroutineScope + Job()).apply {
	connectionScope = this
	launch {
		connection = socketConnectionFactory.create(
			api.httpClientOptions,
			incomingMessages,
			coroutineContext,
		).connectAndBind(this)
	}
}
```

Reply:
Sorry, was distracted and didn't reply in time. The `also` reads a bit bloaty too.
How about

```kotlin
val scope = coroutineScope + Job()
scope.launch {
	connection = socketConnectionFactory.create(
		api.httpClientOptions,
		incomingMessages,
		coroutineContext,
	).connectAndBind(this)
}
connectionScope = scope
```